x-pack/plugins/spaces/public/space_list/space_list_internal.test.tsx

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { mountWithIntl } from '@kbn/test/jest';
+import { act } from '@testing-library/react';
+import { coreMock } from 'src/core/public/mocks';
+import type { Space } from 'src/plugins/spaces_oss/common';
+import type { SpaceListProps } from '../../../../../src/plugins/spaces_oss/public';
+import { getSpacesContextWrapper } from '../spaces_context';
+import { spacesManagerMock } from '../spaces_manager/mocks';
+import { ReactWrapper } from 'enzyme';
+import { SpaceListInternal } from './space_list_internal';
+
+const ACTIVE_SPACE: Space = {
+  id: 'default',
+  name: 'Default',
+  initials: 'D!', // so it can be differentiated from 'Delta'
+  disabledFeatures: [],
+};
+const getSpaceData = (inactiveSpaceCount: number = 0) => {
+  const inactive = ['Alpha', 'Bravo', 'Charlie', 'Delta', 'Echo', 'Foxtrot', 'Golf', 'Hotel']
+    .map<Space>((name) => {
+      const id = name.toLowerCase();
+      return { id, name, disabledFeatures: [`${id}-feature`] };
+    })
+    .slice(0, inactiveSpaceCount);
+  const spaces = [ACTIVE_SPACE, ...inactive];
+  const namespaces = spaces.map(({ id }) => id);
+  return { spaces, namespaces };
+};
+
+/**
```

Comment:
__note:__ it feels like the readers can also benefit from having this description in the actual component file as well